### PR TITLE
Remove Unnecessary Function from readdbc Module

### LINKED ISSUE #9: Update README

### LINKED ISSUE #17: Update pyreaddbc to Remove Unnecessary Functions
**Description:**
With the recent changes made to the pyreaddbc package, an unnecessary function was identified in the readdbc module. This issue is created to track the removal of this function for better code organization and maintainability.

### PR DESCRIPTION
**Pull request by esloch**
**Description:**
This pull request addresses Issues:
Closes #17.
Closes #9.

**Changes Made:**
- Removed the unnecessary function from the readdbc module.
  - [x] read_dbc
  - [x] read_dbc_dbf
  - [x] dbf_to_csvgz
- Enhanced and refactored the test suite.
- Improved the README.
  - [x] Updated the README to provide clearer examples and instructions.
  - [x] Moved the functions removed from the readdbc module to development tests in CI.
  - [x] Moved pandas to the "group dev" in pyproject.toml for testing purposes.
  - [x] Exemplified the use cases of the pyreaddbc package in the README.

**Testing Done:**
- Updated and expanded test cases to ensure the changes don't introduce any regressions.

### CODE REVIEW

**Comment by fccoelho** [comment]
May we could  go to full simplicity, removing the dependency on pandas and just saving the DBF as output.
What do you think @esloch ?

**Review by fccoelho — APPROVED** [review]

**Comment by esloch** [comment]
> May we could go to full simplicity, removing the dependency on pandas and just saving the DBF as output. What do you think @esloch ?

@fccoelho, could you please review the README to ensure that the license information at the end of the file meets our requirements?

File: README.md [hunk 1]
```diff
@@ -1 +1,88 @@
-# py-blast-dbf
+# pyreaddbc
+
+**pyreaddbc** is a Python library for working with DBC (Dengue Brazilian Case Registry) and DBF (dBase) files. It is an extension used in the PySUS package, which provides access to Brazilian public health data. For more information, you can visit the [PySUS repository](https://github.com/AlertaDengue/pysus).
```
[the remark this answers is not in the document]
Comment (fccoelho) [reply README.md:3]: Replace the second sentence by: 
"Legacy systems from the Brazilian Ministry of Health still uses DBF and DBC formats to Publicize data. This package was developed to help PySUS  to extract data from these formats to more modern  ones. Pyreaddbc can also be used to convert DBC files from any other source.

File: README.md [hunk 2]
```diff
@@ -1 +1,88 @@
...
+
+```bash
+pip install pyreaddbc
+```
+
+## Usage
+
+### Reading DBC Files
+
+To read a DBC file and convert it to a pandas DataFrame, use the `read_dbc` function:
```
Comment (fccoelho) [inline README.md:17]: Add this sentence:  "Extracting the DBF from a DBC may require to specify the encoding of the original data, if known."

**Review by fccoelho — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (esloch) [reply README.md:20]: Done!

Comment (esloch) [reply README.md:20]: Thank you for your review @fccoelho!. I am going to merge this Pull Request. 🚀

**Review by fccoelho — APPROVED** [review]

**Pull request merged**